catkin_ws/src/cucumber_msgs/msg/Cucumber.msg

```diff
@@ -5,18 +5,20 @@ Header header
 byte dimension
 
 # The x,y and z position of the stem of the produce.
-# NOTE: When 'dimension' = 2D, the z value should be omitted.
+# NOTE: When 'dimension' = DIM_2D, these values should be omitted.
 geometry_msgs/Vector3 stem_position
 
-# The width and height of the produce.
-# When 'dimension' = 2D, these values are between 0 and 1.
-# When 'dimension' = 3D, these values are real world positions in meter.
+# The width and height of the produce in meter.
+# When 'dimension' = 2D, these values are in pixels.
 float32 width
 float32 height
 
+# The x,y position of the stem in the original image it was detected in.
+int32[] image_stem_position
```

Comment:
Too bad there's no 2d point object in ros, it would definitely be a lot nicer to use that...